Watchman.Discord/Areas/Reactions/Services/ReactionsService.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Threading.Tasks;
+using Discord.WebSocket;
```

Comment:
nie możesz używać Discord.WebSocket ani nic z biblioteki Discord

Reply:
Discord.NET*

Reply:
a to dlaczego nie można?

Reply:
bo założenie mamy takie, że chcemy być niezależni od biblioteki Discord.NET
załóżmy że biblioteka przestała być wspierana i co wtedy? cały projekt do kosza i od nowa bo mamy pełno zależności?
dzięki temu że mamy swój framework, wystarczy podmienić kilka rzeczy i całość działa

Reply:
albo co gdyby pojawiła sie lepsza biblioteka?
lepiej przepiąć kod w 5 albo 6 miejscach
niż przepisywać większość projektu, bo na każdym kroku mamy zależności 

albo co gdybyśmy chcieli napisać własną biblioteke, żeby mieć lepszą kontrole nad tym co sie dzieje pod spodem?

sporo na wydajności można zyskać w taki sposób
w pracy pisali od zera własną implementacje locka, bo okazało sie że ten wbudowany w c# jest za wolny
tak samo jak pisali od zera własnego ORM, bo te gotowe... one są dobre, ale nie jak chcesz wycisnąć max moc

tutaj na wydajności (poza kilkoma miejscami) aż tak nam nie zależy
ale mimo wszystko własna biblioteka do obsługi api dawałaby nam więcej informacji
niż takie discord.net
które niby jest open source
ale nie mamy w zespole kogoś, kto dokładnie wiedziałby jak ten system działa pod spodem

w przypadku naszego kodu... co prawda nie mamy osoby która zna cały projekt 
ale każdy z nas dobrze ogarnia pewne mechanizmy, bo sam je napisał
i jest w stanie wykryć że jeśli psuje sie pewna rzecz na pewnym etapie
to prawdopodobnie problem jest w X miejscu

no i jeszcze kwestia samego wsparcia
może być tak, że discord kiedyś zmieni swoje api 
i jakaś funkcjonalność nie będzie działać, bo discord.net nie będzie dostosowane

obecnie discord.net używamy, bo nie mamy czasu na napisanie czegoś własnego 
ale nie jest to oficjalna biblioteka
nie mamy pewności że zawsze będzie działać i będzie zawsze aktualna 

https://github.com/discord-net/Discord.Net 
tu możesz spojrzeć na ich kod